src/app/dataset-view/additional-components/flows-component/components/flows-table/flows-table.component.html

```diff
@@ -0,0 +1,169 @@
+<table mat-table [dataSource]="dataSource">
+    <!-- Description Column -->
+    <ng-container matColumnDef="description">
+        <th mat-header-cell *matHeaderCellDef>
+            <div class="d-inline-block text-muted">
+                <button mat-button [matMenuTriggerFor]="statusMenu">
+                    <span class="text-dark"> Status <span class="carpet"></span> </span>
+                </button>
+                <mat-menu #statusMenu="matMenu" xPosition="after" [hasBackdrop]="true">
+                    <button
+                        mat-menu-item
+                        [class.selected-status]="!filterByStatus"
+                        (click)="changeFilterByStatus(null)"
+                    >
+                        <span class="text-small"> All </span>
+                    </button>
+                    <mat-divider></mat-divider>
+
+                    <ng-container *ngFor="let status of FlowStatus | keyvalue">
+                        <button
+                            mat-menu-item
+                            [class.selected-status]="filterByStatus === status.value"
+                            (click)="changeFilterByStatus(status.value)"
+                        >
+                            <span class="text-small">
+                                {{ status.value | lowercase | titlecase }}
+                            </span>
+                        </button>
+                        <mat-divider></mat-divider>
+                    </ng-container>
+                </mat-menu>
+            </div>
+        </th>
+        <td mat-cell *matCellDef="let element" style="height: 62px">
+            <div class="d-flex py-1">
+                <mat-icon class="icon-status margin-top-5" [ngClass]="descriptionColumnOptions(element).class">{{
+                    descriptionColumnOptions(element).icon
+                }}</mat-icon>
+                <div class="ms-1 py-2">
+                    <span class="fw-500">
+                        {{ descriptionForDatasetFlow(element) }}
+                        <span>{{ descriptionDatasetFlowEndOfMessage(element) }}</span>
+                    </span>
+                    <br />
+                    <span class="text-small truncate-sub-message d-block">{{
+                        descriptionDatasetFlowSubMessage(element, fetchStep!)
```

Comment:
No need to propagate `fetchStep` like this, there is component state.
An, in fact, this would fail for Transform flow, since there is no `fetchStep` there.

Reply:
Fixed